sdk/web-pubsub/web-pubsub-client/review/web-pubsub-client.api.md

```diff
@@ -246,8 +259,10 @@ export class WebPubSubClient {
     on(event: "server-message", listener: (e: OnServerDataMessageArgs) => void): void;
     on(event: "group-message", listener: (e: OnGroupDataMessageArgs) => void): void;
     on(event: "rejoin-group-failed", listener: (e: OnRejoinGroupFailedArgs) => void): void;
+    sendEvent(eventName: string, content: JSONTypes | ArrayBuffer, dataType: WebPubSubDataType, options: SendEventFireAndForgetOptions): Promise<void>;
     sendEvent(eventName: string, content: JSONTypes | ArrayBuffer, dataType: WebPubSubDataType, options?: SendEventOptions): Promise<WebPubSubResult>;
```

Comment:
One practical trouble with overloaded signatures like this one is that you can't pass a value that has a natural `boolean` type to `options.fireAndForget` in TypeScript whereas you can in JavaScript.

Try something like:

```ts
const fireAndForget: boolean = true;

// There will be an error on this line.
await client.sendEvent(..., { fireAndForget });
```

This might be okay, but it means that you can't call this function with a dynamic value of fireAndForget in TypeScript, whereas you can in JavaScript. In TypeScript it has to be known to be either exactly true or exactly false.

Reply:
Your concern makes sense. And users may use in such scenario nowadays. I made updates the PR to minimize such breaking changes.